File: source/src/main/java/org/rss/tools/mpl/domain/element/LineHeader.java
package org.rss.tools.mpl.domain.element;

import org.rss.tools.mpl.domain.Element;

public class LineHeader implements Element {

	private String value;
	
	private Integer level;

	public String getValue() {
		return value;
	}

	public void setValue(String value) {
		this.value = value;
	}

	public Integer getLevel() {
		return level;
	}

	public void setLevel(Integer level) {
		this.level = level;
	}

	@Override
	public String toString() {
		return "LineHeader [" + level + ", " + value + "]";
	}
	
	
}

File: source/src/main/kotlin/org/rss/tools/mpl/domain/element/LineHeader.kt
package org.rss.tools.mpl.domain.element

import org.rss.tools.mpl.domain.Element
//import kotlin.reflect.KProperty

/*class LevelDelegate {

    private var level: Int = 1

    *//**
      Prevents values out of range
     *//*
    operator fun setValue(thisRef: LineHeader, property: KProperty<*>, value: Int){
        level = Math.max(value, 1)
        level = Math.min(level, 6)
    }

    operator fun getValue(thisRef: LineHeader, property: KProperty<*>): Int = level
}*/

data class LineHeader (var value: String, private val _level:Int = 1) : Element {

    var level = _level
    set(value) {
        field = Math.max(value, 1)
        field = Math.min(field, 6)
    }

    override fun toString(): String {
        return "LineHeader [$level, $value]"
    }

}

